### ISSUE TITLE
HowTo: Install/Run Gix-IDE under Debian 11

### ISSUE THREAD

**Issue by dev4inno** [mridoni/gix#97]
Hello Marco,

not sure, if this is the right place ...

... I have prepared a short HowTo (Workaround) for installing Gix-IDE under Debian 11 (see attached).

If it should be put somewhere else, please feel free to move the document to that place.

Ciao
Klaus
[Debian 11 gix-ide.pdf](https://github.com/mridoni/gix/files/8459355/Debian.11.gix-ide.pdf)

**Comment by GitMensch** [comment]
Hm, looks like your commands removed the mysql and postgres options (the pg client should be available in debian, possibly also with the original name).
And when you already modify the .deb file - why not replacing the "wrong qt" dependencies with the right one there?

It would definitely be better for dependency tracking to have those in there, instead of being installed manually.

Would you consider to change the .pdf to either .MD or also upload the original (odt?) file?

**Comment by GitMensch** [comment]
@mridoni: what debian version is your deb file currently targeted for?

**Comment by dev4inno** [comment]
> Hm, looks like your commands removed the mysql and postgres options (the pg client should be available in debian, possibly also with the original name). And when you already modify the .deb file - why not replacing the "wrong qt" dependencies with the right one there?
> 
> It would definitely be better for dependency tracking to have those in there, instead of being installed manually.
> 
> Would you consider to change the .pdf to either .MD or also upload the original (odt?) file?

Can I send the .odt file to you by mail? I guess the file is too large to upload it (I had already problems with the pdf-file).
And: I have removed only the mysql dependencies. You should be able to use PostgreSQL and unixodbc.

**Comment by mridoni** [comment]
> @mridoni: what debian version is your deb file currently targeted for?

Specifically Ubuntu 20.04 (bullseye/sid)

Thanks

**Comment by dev4inno** [comment]
Sorry for the delay - I had to work for my client the last days.

I tried it on my Debian 11 box, but unfortunately there is still a dependancy that is not satisfiable - it's libmysqlclient21:

![Package Installer - gix-ide_001](https://user-images.githubusercontent.com/63194871/206520134-1cc5b234-c4f4-4ea1-a8f1-d9cc70e71647.png)


So my _"hack"_ remains partly in place.

**Comment by mridoni** [comment]
Hello, the packages are Ubuntu-specific, so there probably a small difference in library versions somewhere (Ubuntu versions, as you probably know, do not map *exactly* to Debian versions. I'll try to also generate Debian packages for the next release, it shouldn't be too hard. 

Thanks for the feedback,

**Comment by GitMensch** [comment]
@dev4inno previously you just completely dropped the mysql client. Can you please check if it works when replacing libmyqlclient21 by libmysql++3v5 ? This seems to exist both in Debian and Ubuntu and includes via its depdendencies the necessary libmysql; if this works then @mridoni may switch to that.

**Comment by dev4inno** [comment]
@GitMensch just tried: if I replace libmyqlclient21 by libmysql++3v5 in the "Depends" section the package will install.

**Comment by mridoni** [comment]
> @GitMensch just tried: if I replace libmyqlclient21 by libmysql++3v5 in the "Depends" section the package will install.

That's a different package, that brings in `libmysqlpp` (and not `libmysqlcient`). It works because `libmysql++3v5` has the same package name on Ubuntu and Debian and in turn its dependency chain brings in the right `libmysqlclient` for each distribution. As said, it works, but the right version of `libmysqlcient` should be used instead.

**Comment by GitMensch** [comment]
> It works because `libmysql++3v5` has the same package name on Ubuntu and Debian and in turn its dependency chain brings in the right `libmysqlclient` for each distribution.

Yes, that is the right explanation for my hacky "does work" solution :-)

If this hack leads to "one deb package less" I'd do it that way personally - but I'm ok with whatever you want to do here :-)

**Comment by mridoni** [comment]
The Debian 10 package is (almost) ready and it will be part of the next release. It installs without a problem and it just needs some more tests (Debian 10 does not provide a package for GnuCOBOL 3.1.2, so I have to build it first and at least test the debugger). The Debian 11 package should be much easier: I didn't know how old were some libraries shipped with Debian 10, to the point that I had to bundle a version of nng that is only used in this specific case. 

For those who want to have a look, all the material for the Debian 10 package (the GitHub Actions workflow file and the updated version of the packaging script - `mkdeb.sh` - and the associated control file - `control-debian-10.tpl`) are already available.
